core/server/models/post.js

```diff
@@ -234,11 +233,11 @@ Post = ghostBookshelf.Model.extend({
         if (mobiledoc) {
             this.set('html', utils.mobiledocConverter.render(JSON.parse(mobiledoc)));
         } else {
-            // legacy showdown mode
-            this.set('html', legacyConverter.makeHtml(_.toString(this.get('markdown'))));
+            // legacy markdown mode
+            this.set('html', legacyConverter.render(_.toString(this.get('markdown'))));
```

Comment:
This [test](https://github.com/TryGhost/Ghost/blob/master/core/test/integration/model/model_posts_spec.js#L433) hasn't failed before, because the result of showdown and markdown-it is different.
Showdown wraps the whole markdown into an outer `<p>` tag and that's why `hasChanged('html')` was true for showdown, but false for markdown-it.